Avoids following GCC 12 error.

picture.c: In function 'CreateSolidPicture':
picture.c:876:26: error: array subscript 'union _SourcePict[0]' is partly outside array bounds of 'unsigned char[16]' [-Werror=array-bounds]
  876 |     pPicture->pSourcePict->type = SourcePictTypeSolidFill;
      |                          ^~
picture.c:870:45: note: object of size 16 allocated by 'malloc'
  870 |     pPicture->pSourcePict = (SourcePictPtr) malloc(sizeof(PictSolidFill));
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
picture.c: In function 'CreateLinearGradientPicture':
picture.c:908:26: error: array subscript 'union _SourcePict[0]' is partly outside array bounds of 'unsigned char[32]' [-Werror=array-bounds]
  908 |     pPicture->pSourcePict->linear.type = SourcePictTypeLinear;
      |                          ^~
picture.c:901:45: note: object of size 32 allocated by 'malloc'
  901 |     pPicture->pSourcePict = (SourcePictPtr) malloc(sizeof(PictLinearGradient));
      |                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
picture.c: In function 'CreateConicalGradientPicture':
picture.c:991:26: error: array subscript 'union _SourcePict[0]' is partly outside array bounds of 'unsigned char[32]' [-Werror=array-bounds]
  991 |     pPicture->pSourcePict->conical.type = SourcePictTypeConical;
      |                          ^~

--- tigervnc-1.10.1/unix/xserver/render/picture.c
+++ tigervnc-1.10.1/unix/xserver/render/picture.c
@@ -22,6 +22,8 @@
  * Author:  Keith Packard, SuSE, Inc.
  */
 
+#pragma GCC diagnostic warning "-Warray-bounds"
+
 #ifdef HAVE_DIX_CONFIG_H
 #include <dix-config.h>
 #endif
